scripts/deploy.ts

```diff
@@ -110,6 +178,90 @@ async function main() {
   )
 }
 
+const legacyContractAddress = '0x9e7A15E77e5E4f536b8215aaF778e786005D0f8d'
+
+async function downloadData(
+  factory: OBSSStorage__factory,
+  provider: JsonRpcProvider,
+  signer: Signer
+) {
+  const legacyContract = factory
+    .attach(legacyContractAddress)
+    .connect(provider)
+    .connect(signer)
+
+  console.log(legacyContract.address)
+  const totalFeeds = await legacyContract.lastFeedId()
+  console.log(`Total feeds count: ${totalFeeds.toNumber()}`)
+  const legacyPosts: OBSSStorage.LegacyPostStruct[] = []
+  const legacyReactions: OBSSStorage.LegacyReactionStruct[] = []
+  for (let i = 0; i < 10; i++) {
+    const postsInFeed = await legacyContract.lastFeedPostIds(i)
+    if (postsInFeed.toNumber() === 0) continue
+
+    const feedPosts = await legacyContract.getFeedPosts(
+      i,
+      0,
+      postsInFeed.toNumber()
+    )
+    console.log(`Feed ID: ${i}`)
+    feedPosts.forEach(async (post: OBSSStorage.PostStructOutput, j) => {
+      legacyPosts.push({
+        post: {
+          author: post.author,
+          metadata: {
+            digest: post.metadata.digest,
+            hashFunction: post.metadata.hashFunction,
+            size: post.metadata.size,
+          },
+          commentsFeedId: 0,
+          timestamp: post.timestamp.toNumber(),
+        },
+        feedId: j,
+      })
+      // Collect reactions
+      try {
+        const reactionId = await legacyContract.lastReactionIds(
+          post.metadata.digest
+        )
+        const reaction = await legacyContract.reactions(
+          post.metadata.digest,
+          reactionId
+        )
+        if (
+          reaction.reactionOwner !==
+          '0x0000000000000000000000000000000000000000'
+        ) {
+          legacyReactions.push({
+            reaction: {
+              value: 0,
+              reactionOwner: reaction.reactionOwner,
+              reactionType: reaction.reactionType,
+            },
+            post: {
+              metadata: {
+                digest: post.metadata.digest,
+                hashFunction: post.metadata.hashFunction,
+                size: post.metadata.size,
+              },
+              author: '0x0000000000000000000000000000000000000000',
+              commentsFeedId: 0,
+              timestamp: 0,
+            },
+          })
+        }
+      } catch (_) {
```

Comment:
`catch (e) { console.error (e) }` is better, it will give you some info on what failed